config/_templates/dataset/carla_garage/town10.yaml

```diff
@@ -0,0 +1,353 @@
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_3_route0_11_08_07_34_15',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_9_route0_11_08_02_41_48',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_33_route0_11_08_07_13_02',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_6_route0_11_09_12_05_44',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_27_route0_11_07_23_53_18',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_23_route0_11_09_04_12_10',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_31_route0_11_09_11_15_02',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_13_route0_11_08_05_25_23',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_45_route0_11_08_05_11_11',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_24_route0_11_09_04_25_55',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_20_route0_11_08_13_06_17',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_43_route0_11_09_11_41_50',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_18_route0_11_08_00_26_39',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_10_route0_11_09_03_15_46',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_4_route0_11_09_09_43_46',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_34_route0_11_09_09_09_27',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_26_route0_11_09_03_45_06',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_44_route0_11_09_08_35_28',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_38_route0_11_08_00_52_06',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_22_route0_11_09_11_45_40',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_11_route0_11_09_11_33_42',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_35_route0_11_09_03_04_41',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_14_route0_11_08_21_32_12',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_5_route0_11_08_19_25_05',
+#@  'VehicleTurningRoute/Town10_Rep0_Town10HD_Scenario4_2_route0_11_09_00_28_15',
+#@ ]
+#@ cameras = [
+#@     'rgb',
+#@ ]
+---
+_target_: rbyte.Dataset
+_recursive_: false
+_convert_: all
+sources:
+  #@ for input_id in drives:
+  (@=input_id@):
+    #@ for source_id in cameras:
+    (@=source_id@):
+      index_column: meta._idx_
+      source:
+        _target_: rbyte.io.PathTensorSource
+        path: '${paths.data}/(@=input_id@)/(@=source_id@)/{:04d}.jpg'
+        decoder:
+          _target_: simplejpeg.decode_jpeg
+          _partial_: true
+          colorspace: rgb
+          fastdct: true
+          fastupsample: true
+
+    #@ end
+  #@ end
+
+samples:
+  inputs:
+    #@ for input_id in drives:
+    (@=input_id@):
+      measurements_path: ${paths.data}/(@=input_id@)/measurements.json
+      waypoints_path: ${paths.data}/(@=input_id@)/waypoints.json
+    #@ end
+
+  executor:
+    _target_: concurrent.futures.ThreadPoolExecutor
+
+  storage: dict
```

Comment:
not sure if dataset init is a bottleneck at all in this case but could probably be sped up, see https://github.com/yaak-ai/rmind/blob/main/config/_templates/dataset/yaak/train.yaml#L723-L731